diff --git a/spinand.c b/spinand.c
index ab7ce2e..727177f 100644
--- a/spinand.c
+++ b/spinand.c
@@ -159,6 +159,9 @@ static const struct spinand_info_t spinand_infos[] = {
 	/* FORESEE */
 	{ "F35SQA001G",      SPINAND_ID(0xCD, 0x71, 0x71), 2048,  64,  64, 1024, 1, 1 },
 	{ "F35SQA002G",      SPINAND_ID(0xCD, 0x72, 0x72), 2048,  64,  64, 2048, 1, 1 },
+
+	/* ZETTA */
+	{ "ZD35Q1GC-IB",     SPINAND_ID(0xCD, 0x71), 	   2048,  64,  64, 1024, 1, 1 },
 };
 
 static inline int spinand_info(struct xfel_ctx_t * ctx, struct spinand_pdata_t * pdat)
